examples/bri-3/src/bri/communication/api/messages.controller.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { CommandBus, QueryBus } from '@nestjs/cqrs';
+import { CreateBpiMessageCommand } from '../capabilities/createBpiMessage/createBpiMessage.command';
+import { DeleteBpiMessageCommand } from '../capabilities/deleteBpiMessage/deleteBpiMessage.command';
+import { GetBpiMessageByIdQuery } from '../capabilities/getBpiMessageById/getBpiMessageById.query';
+import { UpdateBpiMessageCommand } from '../capabilities/updateBpiMessage/updateBpiMessage.command';
+import { CreateBpiMessageDto } from './dtos/request/createBpiMessage.dto';
+import { UpdateBpiMessageDto } from './dtos/request/updateBpiMessage.dto';
+import { BpiMessageDto } from './dtos/response/bpiMessage.dto';
+
+@Controller('messages')
+export class MessageController {
+  constructor(private commandBus: CommandBus, private queryBus: QueryBus) {}
+
+  @Get('/:id')
+  async getBpiMessageById(@Param('id') id: string): Promise<BpiMessageDto> {
+    return await this.queryBus.execute(new GetBpiMessageByIdQuery(id));
+  }
+
+  @Post()
+  async createBpiMessage(
+    @Body() requestDto: CreateBpiMessageDto,
+  ): Promise<string> {
+    return await this.commandBus.execute(
+      new CreateBpiMessageCommand(
+        requestDto.id,
+        requestDto.fromId,
+        requestDto.toId,
+        requestDto.content,
+        requestDto.signature,
+        requestDto.type,
+      ),
+    );
+  }
+
+  @Put('/:id')
+  async updateBpiMessage(
+    @Param('id') id: string,
+    @Body() requestDto: UpdateBpiMessageDto,
+  ): Promise<void> {
+    return await this.commandBus.execute(
```

Comment:
ideally, update shall return the updated object. Why are we returning Void?

Reply:
It is really a question of preference. By standard, HTTP PUT cano both return nothing or return the payload. Since we are not returning in all four controllers implemented so far, i would also avoid sending here. If we in the future agree we want to have this we can refactor everywhere at once.